src/classes/ADDR_Addresses_TEST.cls

```diff
@@ -517,7 +517,304 @@ public with sharing class ADDR_Addresses_TEST {
+        }
+    }
+
+    @isTest
+    private static void newUndeliverableAddressShouldSyncToAccountAndContacts() {
+
+        Test.startTest();
+            createTestData(2, 2, true);
+        Test.stopTest();
+
+        Set<Id> accountIds = (new Map<Id,Account>(testAccounts)).keySet();
+        Map<Id, Account> updatedAccountsById = new Map<Id, Account>(
+            [SELECT Undeliverable_Address__c, (SELECT is_Address_Override__c, Undeliverable_Address__c FROM Contacts)
+             FROM Account
+             WHERE Id IN :accountIds]);
+
+        for (Address__c address: testAddresses) {
+            Account updatedAccount = updatedAccountsById.get(address.Household_Account__c);
+            System.assertEquals(address.Undeliverable__c,
+                    updatedAccount.Undeliverable_Address__c, 'account address' +
+                            ' undeliverable status should not match the address undeliverable status when the address' +
+                            ' is not marked as the default address.');
+            for (Contact contact : updatedAccount.Contacts) {
+                System.assert(contact.is_Address_Override__c == false, 'The contact should not be marked with an ' +
+                        'address override since that will cause inaccuracies in this test scenario.');
+                System.assertEquals(address.Undeliverable__c, contact.Undeliverable_Address__c, 'The contact ' +
+                        'undeliverable address status should match the default address undeliverable status when the ' +
+                        'contact address override value is false.');
+            }
+        }
+    }
+
+    @isTest
+    private static void updatedUndeliverableAddressShouldSyncToAccountAndContacts() {
+        createTestData(3, 3);
+
+        Test.startTest();
+            for (Address__c address : testAddresses) {
+                address.Undeliverable__c = true;
+            }
+            update testAddresses;
+        Test.stopTest();
+
+        Set<Id> accountIds = (new Map<Id,Account>(testAccounts)).keySet();
+        Map<Id, Account> updatedAccountsById = new Map<Id, Account>(
+        [SELECT Undeliverable_Address__c, (SELECT is_Address_Override__c, Undeliverable_Address__c FROM Contacts)
+            FROM Account
+            WHERE Id IN :accountIds]);
+
+        for (Address__c address: testAddresses) {
+            Account updatedAccount = updatedAccountsById.get(address.Household_Account__c);
+            System.assertEquals(address.Undeliverable__c,
+                    updatedAccount.Undeliverable_Address__c, 'account address' +
+                            ' undeliverable status should not match the address undeliverable status when the address' +
+                            ' is not marked as the default address.');
+            for (Contact contact : updatedAccount.Contacts) {
+                System.assert(contact.is_Address_Override__c == false, 'The contact should not be marked with an ' +
```

Comment:
The message associated with this assertion seems more like a code comment than something appropriate for a message assertion.  I'd recommend just stating what the assertion expects and if you feel the need for a comment to explain the reason for the test, do that separately.

Reply:
updated!